coreUtilities/src/main/java/org/opensearch/migrations/transformation/CanApplyResult.java

```diff
@@ -0,0 +1,27 @@
+package org.opensearch.migrations.transformation;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * The result after checking if a transformer can be applied to an entity
+ */
+public abstract class CanApplyResult {
+    public final static CanApplyResult YES = new Yes();
+    public final static CanApplyResult NO = new No();
```

Comment:
why not use an enum for this?
CanApply or CanApplyEnum might be quicker to grok.  I had thought that you were indicating if a result could be applied (I admit that in retrospect, that's dumb, but I may not be the only one).